data/pigui/modules/new-game-window/recovery.lua

```diff
@@ -30,23 +30,37 @@ local fixupDoc = Helpers.versioned {{
 
 		local refs = {}
 
+		local function unpickle(tbl)
+			local result = {}
+			tbl = tbl.table
+			for i = 1, #tbl, 2 do
+				result[tbl[i]] = tbl[i + 1]
+			end
+			return result
+		end
+
+		-- get refs and unpickle 1 level
 		local function getRefs(tbl)
 			if type(tbl) ~= 'table' then return end
 			if tbl.ref and tbl.table and not refs[tbl.ref] then
-				refs[tbl.ref] = tbl.table
+				refs[tbl.ref] = unpickle(tbl)
 			end
 			for _, v in pairs(tbl) do
 				getRefs(v)
 			end
 		end
 
-		local function putRefs(tbl)
+		local function putRefs(tbl, cache)
 			if type(tbl) ~= 'table' then return end
-			for _, v in pairs(tbl) do
-				putRefs(v)
-			end
-			if tbl.ref and not tbl.table then
-				tbl.table = refs[tbl.ref]
+			if cache[tbl] then return end
+			cache[tbl] = true
+			for k, v in pairs(tbl) do
+				if type(v) == 'table' then
+					if v.ref then
+						tbl[k] = refs[v.ref]
+					end
```

Comment:
Note: serialized keys can also be table instances. The in-progress EquipSet v2 came close to using the pattern of serializing `{ [table] = true }` but eventually was refactored away. I'm unsure if we have other occurrences of that pattern.

Reply:
Indeed, there are such places, although not in the restored parameters.

Reply:
Not in the recovered parameters *right now* at least. This may change at any time.

Reply:
Now I take this into account!